.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+exclude: "GeneratedTorchOps\\.td|abstract_interp_lib_gen\\.py|\\.excalidraw"
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-ast
+    -   id: check-yaml
+    -   id: check-added-large-files
+-   repo: https://github.com/psf/black
+    rev: 22.10.0
+    hooks:
+    -   id: black
```

Comment:
How about using [Ruff](https://github.com/astral-sh/ruff) for formatting and linting Python?

Reply:
I'd prefer to stick with black as we use it on other projects and people/tooling are more familiar with it. I ran it a bunch of times during testing this patch, and performance was not an issue for this project.

Can reevaluate later if it makes a difference.